### PR TITLE
[PW-4938] Add 3-digit padding helper method for shopperReference

### PR DESCRIPTION
**Pull request by RokPopov**
**Description**
Since checkout v66 Shopper reference should be 3 digits and above which means in cases that is less the api returns error. Add a method that pads all the shopper reference before adding it to any request.

**Tested scenarios**
- made sure all the requests where the $shopperReference is relevant, have that variable padded

### CODE REVIEW

File: Model/AdyenInitiateTerminalApi.php [hunk 1]
```diff
@@ -262,7 +262,8 @@ public function addSaleToAcquirerData($request, $quote)
 
             if (!empty($recurringContract) && !empty($shopperEmail)) {
                 $saleToAcquirerData['shopperEmail'] = $shopperEmail;
-                $saleToAcquirerData['shopperReference'] = str_pad((string)$customerId, 3, '0', STR_PAD_LEFT);
+                $saleToAcquirerData['shopperReference'] = $this->adyenHelper->padShopperReference($customerId);
+//                                    str_pad((string)$customerId, 3, '0', STR_PAD_LEFT);
```
Comment (candemiralp) [inline Model/AdyenInitiateTerminalApi.php:266]: Is this comment relevant?

File: Model/Api/PaymentRequest.php [hunk 2]
```diff
@@ -230,12 +238,12 @@ public function disableRecurringContract($recurringDetailReference, $shopperRefe
         }
     }
 
-    /**
-     * @param $shopperReference
-     * @return string
-     */
-    private function getShopperReferencePadding($shopperReference)
```
Comment (candemiralp) [inline Model/Api/PaymentRequest.php:237]: Shall we keep this or delete?

File: Helper/Data.php [hunk 3]
```diff
@@ -1792,4 +1792,9 @@ public function parseTransactionId($transactionId)
             return is_string($index);
         }, ARRAY_FILTER_USE_KEY);
     }
+
+    public function padShopperReference($customerId)
```
Comment (candemiralp) [inline Helper/Data.php:1796]: Let's use the variable name `shopperReference` instead of `customerId`.

File: Helper/Data.php [hunk 4]
```diff
@@ -1793,7 +1793,11 @@ public function parseTransactionId($transactionId)
         }, ARRAY_FILTER_USE_KEY);
     }
 
-    public function padShopperReference($shopperReference)
+    /**
+     * @param $shopperReference
+     * @return string
+     */
+    public function padShopperReference($shopperReference): string
```
Comment (Morerice) [inline Helper/Data.php:1800]: Can we also add the argument type?
```suggestion
    public function padShopperReference(string $shopperReference): string
```

**Review by Morerice — CHANGES_REQUESTED** [review]

**Review by Morerice — APPROVED** [review]

**Review by AlexandrosMor — APPROVED** [review]

**Pull request merged**